.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeParser.cs

```diff
@@ -253,11 +253,14 @@ private DateTimeResolutionResult MergeDateAndTime(string text, DateObject refere
             ret.PastValue = DateObject.MinValue.SafeCreateFromValue(pastDate.Year, pastDate.Month, pastDate.Day, hour, min, sec);
 
             // Handle case like "Wed Oct 26 15:50:06 2016" which year and month separated by time.
-            var matchYear = this.config.YearRegex.Match(text.Substring(er2[0].Start + er2[0].Length ?? 0));
-            if (matchYear.Success && pr1.TimexStr.Substring(0, 4) == "XXXX" && pr1.TimexStr.Substring(5, 2) != "XX")
+            var timeSuffix = text.Substring(er2[0].Start + er2[0].Length ?? 0);
```

Comment:
Can we do some benchmarking to see the perf impact of this change? @STayinloves can maybe guide you.

Reply:
@Grey0202 unfortunately there are now also some conflicts to resolve.

Reply:
> Can we do some benchmarking to see the perf impact of this change? @STayinloves can maybe guide you.

I test this many times and the current version takes about 44270ms and PR version takes about 44314ms for all 10314 test on average (including copy the test for 10 times and run the test for more than 10 times they all get similar results). The average length of the sentence is 31.